azurerm/resource_arm_network_interface.go

```diff
@@ -497,13 +514,21 @@ func flattenNetworkInterfaceIPConfigurations(ipConfigs *[]network.InterfaceIPCon
 		props := ipConfig.InterfaceIPConfigurationPropertiesFormat
 
 		niIPConfig["name"] = *ipConfig.Name
-		niIPConfig["subnet_id"] = *props.Subnet.ID
+
+		if props.Subnet != nil {
```

Comment:
can we also check that `props.Subnet.ID` isn't nil here too (which is possible given a malformed API response)